R/plotAbundanceDensity.R

```diff
@@ -0,0 +1,163 @@
+    # Gets the most abundant taxa
+    top_taxa <- getTopTaxa(object, n)
+    # Subsets abundance table  by taking taxa of highest abundance
+    abundance_table_subset <- abundance_table[top_taxa, , drop=FALSE]
+    
+    # Converts top taxa names to factor
+    top_taxa <- factor(top_taxa, rev(top_taxa))
+    
+    # Creates a data frame that includes sample names, taxa names as factors,
+    # and abundance values
+    density_data <- data.frame(
+        # Sample data = sample 1 is repeated n(taxa) times, sample 2 repeated n(taxaa) times...
+        sample = rep(seq_len(ncol(abundance_table_subset)), each = nrow(abundance_table_subset)), 
+        # taxa data = taxa 1, taxa 2,.. taxa n, taxa 1, taxa 2,... taxa n, as many times as there are samples
+        taxa = rep(top_taxa, ncol(abundance_table_subset)),
+        # Values grom abundance table
+        value = as.numeric(abundance_table_subset)
+    )
+    
+    # Gets coloring information if 'colour_by' is not NULL, otherwise specifies just aesthetic
+    if (!is.null(colour_by)) {
+        # Gets information from colData
+        colour_out <- retrieveCellInfo(object, colour_by)
+        # Mirrors back the variable name, if a partial match was used
+        colour_by <- colour_out$name
+        # Adds information to the data frame that includes all density data
+        density_data$colour_by <- colour_out$val[density_data$sample]
+        # Specifies aesthetic
+        aesthetic <- aes_string(y = "taxa", x = "value", colour = "colour_by")
+    } else {
+        # Specifies aesthetic, when colour_by is not used
+        aesthetic <- aes_string(y = "taxa", x = "value")
+    }
+    
+    # Creates a list that includes the data and aesthetic
+    density_data_list <- list(density_data = density_data, 
+                              aesthetic = aesthetic)
+    return(density_data_list)
+}
+
+.density_plotter <- function(density_data, 
+                             aesthetic,
+                             abund_values,
+                             colour_by,
+                             point_alpha = 0.6,
+                             point_shape = 124,
+                             text_size = 8,
+                             text_colour = "gray35"){
+    # Creates the plot and annotations. 
+    plot_out <- ggplot(density_data, aesthetic) + geom_point(alpha = point_alpha, shape = point_shape)
+    plot_out <- plot_out + xlab(abund_values) + ylab("Feature")  + theme_bw(text_size) +
+        theme(axis.text.x = element_text(colour = text_colour),
+              axis.text.y = element_text(colour = text_colour),
+              axis.title.x = element_text(colour = text_colour),
+              axis.title.y = element_text(colour = text_colour),
+              title = element_text(colour = text_colour))
+    
+    if (!is.null(density_data$colour_by)) {
+        plot_out <- .resolve_plot_colours(plot_out, density_data$colour_by, colour_by)
+    }
+    return(plot_out)
+}
```

Comment:
Have a look at the other plotting functions and use the helper functions like `.resolve_plot_colours`, `.get_point_args`